R/write_man.R

```diff
@@ -0,0 +1,192 @@
+#' to your DESCRIPTION file. If you made a project using rUM this happens automatically.
+#' 
+#' 
+#' @return A \code{.Rd} file in the \code{man} package directory corresponding to the 
+#' name of the supplied dataset.
+#' 
+#' @export
+#' 
+#' @examples
+#' if (interactive()) {
+#'   # Dataset object from Global Environment (unquoted)
+#'   write_man(mtcars)
+#'   
+#'   # Dataset object from Global Environment as character string (quoted)
+#'   write_man("mtcars")
+#' }
+write_man <- function(the_dataset) {
+
+  # Handle both quoted and unquoted dataset names
+  if (is.character(the_dataset) && length(the_dataset) == 1) {
+    # Input is a string:
+    # This allows write_man("mtcars")
+    the_dataset_name <- the_dataset
+    
+    # Check if the dataset exists
+    if (!exists(the_dataset_name, envir = .GlobalEnv)) {
+      stop("Dataset '", the_dataset_name, "' not found in the global environment")
+    }
+    
+    # Get the actual dataset object
+    the_dataset <- get(the_dataset_name, envir = .GlobalEnv)
+  } else {
+    # Input is an unquoted object:
+    # This allows write_man(mtcars)
+    the_dataset_name <- deparse(substitute(the_dataset))
+    
+    # Check if the unquoted dataset exists
+    if (!exists(the_dataset_name, envir = .GlobalEnv)) {
+      stop("Dataset '", the_dataset_name, "' not found in the global environment")
+    }
+  }
+
+  # rUM needs labelled dependencies for labelled::var_label
+  # Check if the dataset exists in the global environment: use the quoted name
+  if (!exists(the_dataset_name, envir = .GlobalEnv)) {
+    stop("Dataset '", the_dataset_name, "' not found in the global environment")
+  }
+  
+  # EDIT: We already have the dataset object in the_dataset parameter, so this section
+  # is being removed.
+  # dataset_obj <- get(the_dataset, envir = .GlobalEnv)
+  
+  # Check if the object is a data.frame or tibble
+  if (!inherits(the_dataset, "data.frame")) {
+    stop("Object '", the_dataset_name, "' is not a data.frame or tibble")
+  }
+  
+  # Check if R folder exists
+  if (!dir.exists("R")) {
+    stop("The R folder does not exist in the current directory")
+  }
+  
+  # Construct the potential file path for the R documentation file
+  # Use the string name of the dataset for the file name
+  r_file_path <- file.path("R", paste0(the_dataset_name, ".R"))
+  
+  # Check if the file already exists
+  if (file.exists(r_file_path)) {
+    stop("Documentation file '", r_file_path, "' already exists")
+  }
+  
+  # Determine if it's specifically a tibble
+  is_tibble <- inherits(the_dataset, "tbl_df")
+  
+  n_rows <- nrow(the_dataset)
+  n_cols <- ncol(the_dataset)
+  
+  n_rows_formatted <-
+    format(n_rows, big.mark = ",", scientific = FALSE, trim = TRUE)
+  n_cols_formatted <-
+    format(n_cols, big.mark = ",", scientific = FALSE, trim = TRUE)
+  
+  # Choose the appropriate data structure label
+  data_structure <- if (is_tibble) "tibble" else "data frame"
```

Comment:
I know this is for a message, but just to make me feel better, can we call this a "data.frame" instead?

Reply:
change data frame to data.frame